src/Microsoft.Identity.Client/Telemetry.cs

```diff
@@ -48,5 +50,129 @@ public static Telemetry GetInstance()
         {
             return Singleton;
         }
+
+        public bool TelemetryOnFailureOnly { get; set; }
+
+        private Dictionary<Tuple<string, string>, EventBase> EventsInProgress = new Dictionary<Tuple<string, string>, EventBase>();
+
+        private Dictionary<string, List<EventBase>> CompletedEvents = new Dictionary<string, List<EventBase>>();
+
+        internal string GenerateNewRequestId()
+        {
+            return Guid.NewGuid().ToString();
+        }
+
+        internal void StartEvent(string requestId, EventBase eventToStart)
+        {
+            EventsInProgress[new Tuple<string, string>(requestId, eventToStart["event_name"])] = eventToStart;
+        }
+
+        internal void StopEvent(string requestId, EventBase eventToStop)
+        {
+            Tuple<string, string> eventKey = new Tuple<string, string>(requestId, eventToStop["event_name"]);
+
+            // Locate the same name event in the EventsInProgress map
+            EventBase eventStarted = null;
+            if (EventsInProgress.ContainsKey(eventKey))
+            {
+                eventStarted = EventsInProgress[eventKey];
+            }
+
+            // If we did not get anything back from the dictionary, most likely its a bug that StopEvent
+            // was called without a corresponding StartEvent
+            if (null == eventStarted)
+            {
+                // Stop Event called without a corresponding start_event.
+                return;
```

Comment:
Believe it or not, I thought about that too, when writing this line. In Android's implmentation, somehow it is as simple as `Logger.warning(TAG, null, "blah blah blah")`. I assume in MSAL dotnet I would need to use `requestContext.Logger.Info("blah blah blah")`, however this singleton Telemetry class does not currently have access to the per-request context. This can probably be solved somehow, eventually.

Based on the business value of this log line VS the remaining of telemetry events, I chose to shift my focus on the event implementation. Feel free to create an issue for this one.